### PR TITLE
Puppeteer/add test for request blocking

### PR DESCRIPTION
**Pull request by simsany**

### CODE REVIEW

File: puppeteer/spec/request-blocking.spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    before(async () => {
+      browser = await puppeteer.launch();
+      page = await browser.newPage();
+      await page.setCacheEnabled(false);
+      await page.goto(BASE_URL);
+      await page.setRequestInterception(true);
+
+      page.on('request', (request) => {
+        if (request.url() === USER_LIST_URL) {
+          interceptedRequest = request;
```
Comment (mgabor18) [inline puppeteer/spec/request-blocking.spec.ts]: You save the request into the variable `interceptedRequest`, but you do not use it anywhere.
This seems like an unnecessary step.

**Review by mgabor18 — COMMENTED** [review]
Please add script to `package.json` to execute this test specifically. (similarly to previous solutions)

File: puppeteer/spec/request-blocking.spec.ts (hunk 1 above)
Comment (simsany) [reply puppeteer/spec/request-blocking.spec.ts]: Alt Text

File: puppeteer/spec/request-blocking.spec.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import puppeteer, { HTTPResponse } from 'puppeteer';
+import { USER_LIST_SELECTOR, OUTPUT_RESPONSE } from '../data/selectors.json';
+import { BASE_URL, USER_LIST_URL } from '../data/constants.json';
+import { expect } from 'chai';
+
+describe('Request Blocking', () => {
+  describe('Blocking the request', () => {
+    let browser: puppeteer.Browser;
+    let page: puppeteer.Page;
+    let interceptedRequest: puppeteer.HTTPRequest;
```
Comment (mgabor18) [inline puppeteer/spec/request-blocking.spec.ts]: This variable never gets any value [:

File: puppeteer/spec/request-blocking.spec.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import { BASE_URL, USER_LIST_URL } from '../data/constants.json';
+import { expect } from 'chai';
+import { resolve } from 'path';
+
+describe('Request Blocking', () => {
+  describe('Blocking the request', () => {
+    let browser: puppeteer.Browser;
+    let page: puppeteer.Page;
+    let errorText: string;
+    let intervalId: NodeJS.Timer;
```
Comment (mgabor18) [inline puppeteer/spec/request-blocking.spec.ts]: This variable could be declared inside the `before` hook, as it is not used anywhere else.

File: puppeteer/spec/request-blocking.spec.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        } else {
+          request.continue();
+        }
+      });
+
+      page.on('requestfailed', (request) => {
+        if (request.url() === USER_LIST_URL) {
+          ({ errorText } = request.failure());
+        }
+        request.continue();
```
Comment (mgabor18) [inline puppeteer/spec/request-blocking.spec.ts]: this `request.continue()` is unnecessary.

**Review by mgabor18 — APPROVED** [review]

**Pull request merged**